adapters/pubmatic/pubmatic.go

```diff
@@ -355,8 +366,25 @@ func (a *PubmaticAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *ad
 		return nil, errs
 	}
 
-	if wrapExt != "" {
-		rawExt := fmt.Sprintf("{\"wrapper\": %s}", wrapExt)
+	//get request ext bidder params
+	if wrapperObj, present := reqExtBidderParams["wrapper"]; present && len(wrapperObj) != 0 {
+		var wrpExt pubmaticWrapperExt
+		err = json.Unmarshal(wrapperObj, &wrpExt)
+		if err == nil {
+			if wrpExt.ProfileID != 0 {
+				wrapperExt.ProfileID = wrpExt.ProfileID
+				wrapperExtSet = true
+			}
+			if wrpExt.VersionID != 0 {
+				wrapperExt.VersionID = wrpExt.VersionID
+				wrapperExtSet = true
+			}
+		}
+	}
+
+	if wrapperExtSet {
+		jsonData, _ := json.Marshal(wrapperExt)
+		rawExt := fmt.Sprintf("{\"wrapper\": %s}", string(jsonData))
```

Comment:
There's always a risk of introducing a json injection attack when manually composing a json blob. Since you're already performing a marshal, consider introducing a wrapper container struct.

Reply:
Fixed.